### PR TITLE
rpc-handler: Fix stale remote server state.

### PR DESCRIPTION
**Pull request by yasserf**

### CODE REVIEW

**Comment by yasserf** [comment]
@ronag

File: src/rpc/rpc-handler.js [hunk 1]
```diff
@@ -130,30 +132,22 @@ module.exports = class RpcHandler {
...
     }
 
-    /*
-  * No local or remote providers to service the request? Return here
-  */
-    if (allRemoteProviderTopics.length === 0) {
-      return null
+    const servers = this._subscriptionRegistry.getAllRemoteServers(rpcName)
+
+    index = Math.round(Math.random() * (servers.length - 1))
```
Comment (ronag) [inline src/rpc/rpc-handler.js]: `Math.floor(Math.random() * servers.length)`

File: src/rpc/rpc-handler.js [hunk 2]
```diff
@@ -121,7 +122,8 @@ module.exports = class RpcHandler {
     }
 
     const subscribers = Array.from(this._subscriptionRegistry.getLocalSubscribers(rpcName))
-    let index = Math.floor(Math.random() * subscribers.length)
+    index = Math.floor(Math.random() * subscribers.length)
+
```
Comment (ronag) [inline src/rpc/rpc-handler.js]: `let index`

File: src/rpc/rpc-handler.js [hunk 3]
```diff
@@ -255,8 +244,11 @@ module.exports = class RpcHandler {
     const correlationId = message.data[1]
     const rpcData = this._rpcs[correlationId]
 
-    if (rpcData.remoteServers && rpcData.remoteServers.length > 0) {
-      const rpcProxy = new RpcProxy(this._options, this._getNextRandomServer(rpcData.remoteServers), rpcName, correlationId)
+    const servers = this._subscriptionRegistry.getAllRemoteServers(rpcName)
+    const server = servers[Math.round(Math.random() * (servers.length - 1))]
```
Comment (ronag) [inline src/rpc/rpc-handler.js]: `const server = servers[Math.floor(Math.random() * servers.length)]`

**Comment by yasserf** [comment]
Thanks @ronag !

**Pull request merged**